projects/vdk-core/src/vdk/internal/cli_entry.py

```diff
@@ -158,16 +161,20 @@ def main() -> None:
     """
     # configure basic logging , it's expected that a plugin would override and set it up properly
     click_log.basic_config(logging.getLogger())
+    plugin_load_success = True
 
     log.debug("Setup plugin registry and call vdk_start hooks ...")
     plugin_registry = PluginRegistry()
     plugin_registry.add_hook_specs(InternalHookSpecs)
-    plugin_registry.load_plugins_from_setuptools_entrypoints()
+    try:
+        plugin_registry.load_plugins_from_setuptools_entrypoints()
+    except Exception:
+        plugin_load_success = False
     plugin_registry.load_plugin_with_hooks_impl(CliEntry(), "cli-entry")
 
     exit_code = cast(InternalHookSpecs, plugin_registry.hook()).vdk_main(
         plugin_registry=plugin_registry,
-        root_command=cli,
+        root_command=cli if plugin_load_success else None,
```

Comment:
in case of plugin load failure, why evaluating the vdk_main at all? It wasn't evaluated before this change (due to the prior exception raised that is now catched)

Reply:
Because we would only log the exception (without logging functionality initialized) and re-raise it, cutting the execution of all vdk-hooks including vdk_exit which writes the termination message and job execution summary, meaning that users wouldn't see any logs in loginsight and the pod would be deleted from k8s as terminated with Error and the data job would be recorded as failed with platform error in the db. The proposed solution would run all other generic hooks, except the vdk command itself in case of such errors as per Toni's suggestion above.

Reply:
In case the vdk-hooks flow is now changed, you should test cover it. Didn't see Toni's suggestion, maybe you could link it?

Reply:
https://github.com/vmware/versatile-data-kit/pull/1241#discussion_r1004526029 . The added unit tests cover new functionality, hit all hooks, including vdk_exit. The only caveat is that I had to use mocks (for the sys.exit and termination message file writer) and introduce the Exceptions as side effects, since it's difficult to introduce this issue with a unit test. The only other test I can think of is to create a specific env with the precise version of clashing dependencies and test exit status codes via bash but I don't like this approach either. N.b the existing functional tests which use CliEntryBasedTestRunner - won't hit the main method in cli_entry.py module.

Reply:
Could you please link the verification of hooks invocation, because I missed it?

Reply:
Also, for `cli_entry.py` coverage maybe can unit test it

Reply:
https://github.com/vmware/versatile-data-kit/blob/62069ddf03322d3a84f2d76b4c1c44758acb74f3/projects/vdk-core/tests/vdk/internal/test_run_load_setuptools_errors.py#L21
This a mocked method in the writer module:
https://github.com/vmware/versatile-data-kit/blob/187853b5a2a812d7cca128ab751f5674da88a13a/projects/vdk-core/src/vdk/internal/builtin_plugins/termination_message/writer.py#L30-L49
which is invoked from the vdk_exit hook. Both unit tests check that it was invoked and that the correct termination message was written e.g Platform or User Error which is the main aim of this change, as previously the termination message wasn't written.

Reply:
I don't see test coverage of hooks being innovated now, oppositely to before the MR change.